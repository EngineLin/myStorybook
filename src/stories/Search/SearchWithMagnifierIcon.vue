<template>
  <div class="search
              flex items-center w-auto bg-gray-200 rounded py-2">

    <div class="search-icon-container
                px-2 flex justify-center items-center">
      <img src="../../assets/images/icons/heroicons/icon-search.svg"
           class="search-icon
                  w-6 h-6 text-gray-200"
           alt="search icon">
    </div>

    <input type="text"
           :value="value"
           @input="onInput"
           placeholder="Search teams or members"
           class="search-input
                  w-full h-full bg-transparent
                  focus:outline-none">
  </div>
</template>

<script>
  import {Component, Vue, Prop, Emit} from 'vue-property-decorator'

  @Component
  export default class SearchWithMagnifierIcon extends Vue {
    @Prop({
      type: String,
      default: ''
    })
    value

    @Emit('change')
    change(value) {
      return value
    }

    onInput(e) {
      const {value} = e.target
      this.change(value)
    }
  }
</script>
