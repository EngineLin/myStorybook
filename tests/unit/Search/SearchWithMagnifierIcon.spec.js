import { shallowMount } from '@vue/test-utils'

import SearchWithMagnifierIcon from '@/stories/Search/SearchWithMagnifierIcon';

it('Match snapshot', () => {
    const wrapper = shallowMount(SearchWithMagnifierIcon, {
        propsData: { value: '' }
    })
    expect(wrapper.html()).toMatchSnapshot()
})

it('Renders props.value when passed', () => {
    const defaultValue = 'default input value'
    const wrapper = shallowMount(SearchWithMagnifierIcon, {
        propsData: {value: defaultValue}
    })
    expect(wrapper.find('.search-input').exists()).toBeTruthy()
    expect(wrapper.find('.search-input').element.value).toEqual(defaultValue)
})

it('Event change invoke when input typing', async () => {
    const inputValue = 'my input typing'
    const wrapper = shallowMount(SearchWithMagnifierIcon, {
        propsData: { value: '' }
    })
    await wrapper.find('.search-input').setValue(inputValue)
    expect(wrapper.emitted().change[0][0]).toEqual(inputValue)
})
