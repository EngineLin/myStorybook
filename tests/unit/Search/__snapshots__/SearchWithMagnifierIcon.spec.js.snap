// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshot 1`] = `
"<div class=\\"search
            flex items-center w-auto bg-gray-200 rounded py-2\\">
  <div class=\\"search-icon-container
              px-2 flex justify-center items-center\\"><img src=\\"../../assets/images/icons/heroicons/icon-search.svg\\" alt=\\"search icon\\" class=\\"search-icon
                w-6 h-6 text-gray-200\\"></div> <input type=\\"text\\" placeholder=\\"Search teams or members\\" class=\\"search-input
                w-full h-full bg-transparent
                focus:outline-none\\">
</div>"
`;
